mm2src/kdf_walletconnect/src/lib.rs

```diff
@@ -0,0 +1,622 @@
+pub mod chain;
+mod connection_handler;
+#[allow(unused)] pub mod error;
+pub mod inbound_message;
+mod metadata;
+#[allow(unused)] mod pairing;
+pub mod session;
+mod storage;
+
+use crate::connection_handler::{handle_disconnections, MAX_BACKOFF};
+use crate::session::rpc::propose::send_proposal_request;
+
+use chain::{WcChainId, WcRequestMethods, SUPPORTED_PROTOCOL};
+use common::custom_futures::timeout::FutureTimerExt;
+use common::executor::abortable_queue::AbortableQueue;
+use common::executor::{AbortableSystem, Timer};
+use common::log::{debug, info, LogOnError};
+use common::{executor::SpawnFuture, log::error};
+use connection_handler::Handler;
+use error::WalletConnectError;
+use futures::channel::mpsc::{unbounded, UnboundedReceiver};
+use futures::StreamExt;
+use inbound_message::{process_inbound_request, process_inbound_response, SessionMessageType};
+use metadata::{generate_metadata, AUTH_TOKEN_DURATION, AUTH_TOKEN_SUB, PROJECT_ID, RELAY_ADDRESS};
+use mm2_core::mm_ctx::{from_ctx, MmArc};
+use mm2_err_handle::prelude::*;
+use pairing_api::PairingClient;
+use relay_client::websocket::{connection_event_loop as client_event_loop, Client, PublishedMessage};
+use relay_client::{ConnectionOptions, MessageIdGenerator};
+use relay_rpc::auth::{ed25519_dalek::SigningKey, AuthToken};
+use relay_rpc::domain::{MessageId, Topic};
+use relay_rpc::rpc::params::session::Namespace;
+use relay_rpc::rpc::params::session_request::SessionRequestRequest;
+use relay_rpc::rpc::params::{session_request::Request as SessionRequest, IrnMetadata, Metadata, Relay,
+                             RelayProtocolMetadata, RequestParams, ResponseParamsError, ResponseParamsSuccess};
+use relay_rpc::rpc::{ErrorResponse, Payload, Request, Response, SuccessfulResponse};
+use serde::de::DeserializeOwned;
+use session::rpc::delete::send_session_delete_request;
+use session::Session;
+use session::{key::SymKeyPair, SessionManager};
+use std::collections::{BTreeSet, HashMap};
+use std::ops::Deref;
+use std::{sync::{Arc, Mutex as SyncMutex},
```

Comment:
nit: we usually call the std mutex just mutex and other async ones async. i like the explicitness in here actually but we better apply it over other instance in the code base for consistency.